dataprofiler/reports/graphs.py

```diff
@@ -80,22 +84,158 @@ def plot_col_histogram(data_type_profiler, ax=None, title=None):
+    ax = sns.histplot(
+        x=histogram['bin_edges'][:-1], bins=histogram['bin_edges'],
+        weights=histogram['bin_counts'], ax=ax)
     if title is None:
         title = str(data_type_profiler.name)
     ax.set_title(title)
     return ax
 
+
+def plot_missing_values_matrix(profiler, ax=None, title=None):
+    """
+    Generates a matrix of bar graphs for the missing value locations within
+    each column in a structured dataset. A color line indicates the value does
+    not exist.
+
+    :param profiler: structured profiler to be plotted
+    :type profiler: StructuredProfiler
+    :param ax: matplotlib axes where to plot the graph
+    :type ax: matplotlib.axes.Axes
+    :param title: title ot set for the graph
+    :type title: str
+    :return: matplotlib figure of where the graph was plotted
+    """
+    if not isinstance(profiler, StructuredProfiler):
+        raise ValueError('`profiler` must of type StructuredProfiler.')
+    return plot_col_missing_values(profiler.profile, ax=ax, title=title)
+
+
+def plot_col_missing_values(col_profiler_list, ax=None, title=None):
+    """
+    Generates a bar graph of the missing value locations within a column where
+    a color line indicates the value does not exist.
+
+    :param col_profiler_list:
+    :type col_profiler_list: list[StructuredColProfiler]
+    :param ax: matplotlib axes where to plot the graph
+    :type ax: matplotlib.axes.Axes
+    :param title: title ot set for the graph
+    :type title: str
+    :return: matplotlib figure of where the graph was plotted
+    """
+    if not (isinstance(col_profiler_list, list)
+            and all(isinstance(col_profile, StructuredColProfiler)
+                    for col_profile in col_profiler_list)):
+        raise ValueError('`col_profiler_list` must be a list of '
+                         'StructuredColProfilers')
+    elif not col_profiler_list:
+        warnings.warn('There was no data in the profiles to plot missing '
+                      'column values.')
+        return
+
+    # bar width settings and height settings for each null value
+    # width = 1, height = 1 would be no gaps
+    width = 0.8
+    height = 1
+
+    # determine the colors for the plot, first color reserved for background of
+    # each column, subsequent colors are for null values.
+    nan_to_color = dict()
+    color_cycle = plt.rcParams['axes.prop_cycle'].by_key()['color'][1:]
+
+    # setup plot
+    is_own_fig = False
+    if ax is None:
+        fig = plt.figure()
+        ax = fig.add_subplot(111)
+        is_own_fig = True
+    # in case user passed their own axes
+    fig = ax.figure
+
+    # loop through eac column plotting their null values
+    for col_id, col_profiler in enumerate(col_profiler_list):
+
+        # plot the values background bar
+        sample_size = col_profiler.sample_size
+        ax.add_patch(matplotlib.patches.Rectangle(
+            xy=(col_id - width / 2 + 0.5, - height / 2),
```

Comment:
does it dynamically change when we get more and more columns?

Reply:
1 would be no gaps between columns, 0 would be the columns wouldn't be visible at all